cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/BrokerSetAwareGoal.java

```diff
@@ -234,7 +232,6 @@ protected void rebalanceForBroker(Broker broker, ClusterModel clusterModel, Set<
                                                        .collect(Collectors.toSet());
       if (maybeApplyBalancingAction(clusterModel, replica, eligibleBrokers, ActionType.INTER_BROKER_REPLICA_MOVEMENT, optimizedGoals,
                                     optimizationOptions) == null) {
-        _provisionResponse = new ProvisionResponse(ProvisionStatus.UNDER_PROVISIONED);
```

Comment:
I dont fully understand. If a brokerSetAware goal is violated and can not be fixed, the recommendation provided by BrokerSetAwareGoal would be UNDER_PROVISIONED, since we would need to add more brokers to the brokerSet. There could be other brokerSets that do not need to add any additional brokers.

Reply:
@mohitpali Yes, it is going to be `UNDER_PROVISIONED`. Please check out how `AbstractGoal#optimize` handles `OptimizationFailureException` -- note that `BrokerSetAwareGoal` extends `AbstractGoal`.

Reply:
Ok, makes sense.